templates/zerver/emails/digest.html

```diff
@@ -31,12 +31,12 @@
 
     <br />
 
-    <p class="digest_paragraph"><a href="{{ realm_uri }}">{% trans %}Click here to log in to Zulip and catch up.{% endtrans %}</a></p>
+    <p class="digest_paragraph"><a href="{{ realm_url }}">{% trans %}Click here to log in to Zulip and catch up.{% endtrans %}</a></p>
 {% endblock %}
 
 {% block manage_preferences %}
 <p class="digest_paragraph">
-    <a href="{{ realm_uri }}/#settings/notifications">{% trans %}Manage email preferences{% endtrans %}</a> |
+    <a href="{{ realm_url }}/#settings/notifications">{% trans %}Manage email preferences{% endtrans %}</a> |
```

Comment:
I think it's possible some of these email templates are broken? It seems like there's still some `realm_uri` usage in `send_email.py`, though maybe that's just for custom emails, it still feels bad to find things like that.

Ideally, the remaining work would be done more systematically, with actual manual testing. Do a clean commit that is just changing the email templates and `zerver/lib/send_email` and the related tests, and another that just changes `zerver/context_processors.py` and the non-email templates that rely on that. Might be good to just make a new PR for those two changes so CI can run on just them, and we can come back to this PR ID for a final sweep. 

And on each commit, you can actually test that the URLs generated are correct.

I think ideally we'd end up with `realm_uri` completely purged except for the ones intentionally present in #30022, which this PR seems to not do.

Reply:
I have opened a #30029 to replace `realm_uri` with `realm_url` in `send_email.py`, email templates, `context_processors.py` and some other related templates and backend files. Other than that and the ones left in #30022, `realm_uri` will be left in the following files:
- Some frontend files
- `zerver/lib/markdown/__init__.py`
- `zerver/lib/push_notifications.py`
- `zerver/views/auth.py`
- `zerver/views/development/dev_login.py`

These can be changed in this PR after #30029 is merged.

Reply:
Yeah that sounds good!